rllib/utils/replay_buffers/utils.py

```diff
@@ -0,0 +1,71 @@
+from ray.rllib.execution.buffers.multi_agent_replay_buffer import (
+    MultiAgentReplayBuffer as LegacyMultiAgentReplayBuffer,
+)
+from ray.rllib.utils.metrics.learner_info import LEARNER_STATS_KEY
+from ray.rllib.utils.replay_buffers import MultiAgentPrioritizedReplayBuffer,\
+    ReplayBuffer
+from ray.rllib.utils.typing import ResultDict, SampleBatchType, TrainerConfigDict
+
+
+def update_priorities_in_replay_buffer(
+    replay_buffer: ReplayBuffer,
+    config: TrainerConfigDict,
+    train_batch: SampleBatchType,
+    train_results: ResultDict,
+) -> None:
+
+    """Updates the priorities in a prioritized replay buffer, given training results.
+
+    The `abs(TD-error)` from the loss (inside `train_results`) is used as new
+    priorities for the row-indices that were sampled for the train batch.
+
+    Args:
+        replay_buffer: The replay buffer, whose priority values to update. This may also
+            be a buffer that does not support priorities.
+        config: The Trainer's config dict.
+        train_batch: The batch used for the training update.
+        train_results: A train results dict, generated by e.g. the `train_one_step()`
+            utility.
+    """
+    # Only update priorities if buffer supports them.
+    if (
+        type(replay_buffer) is LegacyMultiAgentReplayBuffer
+        and config["replay_buffer_config"].get(
+        "prioritized_replay_alpha", 0.0
+    )
+        > 0.0
+    ) or isinstance(
+        replay_buffer, MultiAgentPrioritizedReplayBuffer
+    ):
```

Comment:
raise exception if things are configured incorrectly, instead of silently skip the action?

Reply:
Hence the comment above:

```
    # Only update priorities if buffer supports them.
```